### PR TITLE
[Quant][Inductor] Enable quantization conv_binary(add/add_relu) pattern fusion inside inductor

### PR DESCRIPTION
**Pull request by leslie-fang-intel**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #107951
* #106958
* #106838
* #106836
* #107565
* #105996
* #105906
* #105639
* __->__ #105456
* #105455
* #104590
* #104588
* #104581
* #104580

**Summary**
Enable the `dequant-conv2d-binary_postop(add)-unary_postop(relu)-quant` pattern fusion and lowering inside inductor.

**Test Plan**
```
clear && python -m pytest test_mkldnn_pattern_matcher.py -k test_qconv2d_binary
```

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @ngimel @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov

### CODE REVIEW

File: torch/ao/quantization/pt2e/graph_utils.py [hunk 1]
```diff
@@ -22,7 +22,7 @@
     {torch.nn.ReLU, torch.nn.functional.relu, torch.nn.functional.relu_},
     {torch.nn.BatchNorm2d, torch.nn.functional.batch_norm},
     {torch.nn.Hardtanh, torch.nn.functional.hardtanh, torch.nn.functional.hardtanh_},
-    {torch.add, operator.add, operator.iadd},
+    {torch.add, operator.add, operator.iadd, "add", "add_"},
```
Comment (leslie-fang-intel) [inline torch/ao/quantization/pt2e/graph_utils.py:25]: Hi @kimishpatel, I think we need to extend the `_EQUIVALENT_TYPES` list with "add", "add_" for cases of `x.add(y)` and `x.add_(y)`. Please kindly help to take a look.

Comment (kimishpatel) [reply torch/ao/quantization/pt2e/graph_utils.py:25]: why is add_ here? Exported program should be functional and thus should not have in place variants like add_

Comment (leslie-fang-intel) [reply torch/ao/quantization/pt2e/graph_utils.py:25]: I think here we should configure the `source_fn` of the node: https://github.com/pytorch/pytorch/blob/93e6fc54fac359d0fd5e314847375086c01cedf2/torch/fx/passes/utils/source_matcher_utils.py#L78. So the `source_fn` may include the in place variants like `operator.iadd` which is already in the list.

**Review by jgong5 — APPROVED** [review]

**Comment by leslie-fang-intel** [comment]
Hi @eellison @jansel, could you kindly help to take a look of this PR?

File: test/inductor/test_mkldnn_pattern_matcher.py [hunk 2]
```diff
@@ -454,11 +514,14 @@ def forward(self, x):
...
+        # 5. Qconv2d_add * 1
+        #    [qconv2d_pointwise_default_1, convert_element_type_5, sub_2, mul_5, add_3, mul_6, round_4, add_4,
+        #     clamp_min_3, clamp_max_3, convert_element_type_6]
         self._test_common(
             mod,
             (v,),
-            10,
-            40,
+            11,
+            51,
```
Comment (eellison) [inline test/inductor/test_mkldnn_pattern_matcher.py:524]: In the future it'd be great if we could make the tests more modular. Checking exact # of matches across separate invocations make it harder to detect which part regressed if something fails

**Review by eellison — APPROVED** [review]

File: test/inductor/test_mkldnn_pattern_matcher.py (hunk 2 above)
Comment (leslie-fang-intel) [reply test/inductor/test_mkldnn_pattern_matcher.py:524]: Sure, thanks for the suggestions.

**Comment by jansel** [comment]
I will let @eellison handle reviews on this stack

File: torch/ao/quantization/pt2e/graph_utils.py (hunk 1 above)
Comment (jerryzh168) [reply torch/ao/quantization/pt2e/graph_utils.py:26]: are these needed? and are these actually taking effect?

Comment (leslie-fang-intel) [reply torch/ao/quantization/pt2e/graph_utils.py:26]: yean, I think it's needed for users who writes `x.add(y)` or `x.add_(y)`.

**Pull request closed without merging**